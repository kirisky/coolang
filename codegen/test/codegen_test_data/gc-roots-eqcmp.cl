class A {
 foo:Int;
};

class Main inherits IO {
  main() : Object { {
    if new A = new A then out_string("impossible\n") else out_string("compare done\n") fi;
	out_string("both As should die now\n\n");
	new Object;
  } };
};
