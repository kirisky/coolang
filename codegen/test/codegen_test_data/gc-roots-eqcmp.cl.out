Allocated an object of type: Main
Allocated an object of type: A
Allocated an object of type: A
Allocated a string
compare done
Allocated a string
both As should die now
Freeing an object of type: String
Freeing an object of type: String
Freeing an object of type: A
Freeing an object of type: A
Allocated an object of type: Object
Freeing an object of type: Object
Freeing an object of type: Main
